pep-0570.rst

```diff
@@ -0,0 +1,329 @@
+  by presenting multiple prototypes with varying numbers of
+  arguments. [#SENDFILE]_
+
+* Some functions use *both* of the above approaches. [#RANGE]_ [#ADDCH]_
+
+One more important idea to consider: currently in the documentation
+there's no way to tell whether a function takes positional-only
+parameters.  ``open()`` accepts keyword arguments, ``ord()`` does
+not, but there is no way of telling just by reading the
+documentation that this is true.
+
+====================
+Syntax And Semantics
+====================
+
+From the "ten-thousand foot view", and ignoring ``*args`` and ``**kwargs``
+for now, the grammar for a function definition currently looks like this::
+
+    def name(positional_or_keyword_parameters, *, keyword_only_parameters):
+
+Building on that perspective, the new syntax for functions would look
+like this::
+
+    def name(positional_only_parameters, /, positional_or_keyword_parameters,
+             *, keyword_only_parameters):
+
+All parameters before the ``/`` are positional-only.  If ``/`` is
+not specified in a function signature, that function does not
+accept any positional-only parameters.
+The logic around optional values for positional-only argument
+Remains the same as the one for positional-or-keyword. Once
+a positional-only argument is provided with a default,
+the following positional-only and positional-or-keyword argument
+need to have a default as well. Positional-only parameters that
+don’t have a default value are "required" positional-only parameters.
+Therefore the following are valid signatures::
+
+    def name(p1, p2, /, p_or_kw, *, kw):
+    def name(p1, p2=None, /, p_or_kw=None, *, kw):
+    def name(p1, p2=None, /, *, kw):
+    def name(p1, p2=None, /):
+    def name(p1, p2, /, p_or_kw):
+    def name(p1, p2, /):
+
+Whilst the followings are not::
+
+    def name(p1, p2=None, /, p_or_kw, *, kw):
+    def name(p1=None, p2, /, p_or_kw=None, *, kw):
+    def name(p1=None, p2, /):
+
+==========================
+Full grammar specification
+==========================
+
+A draft of the proposed grammar specification is::
+
+    new_typedargslist:
+      tfpdef (',' tfpdef)* ',' '/' [',' [typedargslist]] | typedargslist
+
+    new_varargslist:
+      vfpdef (',' vfpdef)* ',' '/' [',' [varargslist]] | varargslist
+
+It will be added to the actual typedargslist and varargslist
+but for easier discussion is presented as new_typedargslist and new_varargslist
+
+
+=========================
+Possible implementations
+=========================
+
+----------------------------------
+Full grammar change as in PEP 3102
+----------------------------------
+
+This implementation will involve a full change of the Grammar. This will
+involve following the steps outlined in PEP 306 [#PEP306]_. In addition, other
+steps are needed including:
+
+* Modifying the code object and the function object to be aware of positional
+  only arguments.
+
+* Modifiying `ceval.c` (`PyEval_EvalCodeEx`, `PyEval_EvalFrameEx`...)
```

Comment:
You're using Markdown-style backticks instead of reST double backticks throughout the PEP.